src/neynarAPI/NeynarAPIClient.ts

```diff
@@ -0,0 +1,679 @@
+    fid: number
+  ): Promise<string | undefined> {
+    const response = await this.apis.user.custodyAddress(fid);
+    return response.data.result.custodyAddress ?? undefined;
+  }
+
+  public async fetchUserVerifications(user: {
+    fid: number;
+  }): Promise<VerificationResponseResult | undefined> {
+    const response = await this.apis.verification.verifications(user.fid);
+    return response.data.result;
+  }
+
+  /**
+   * Checks if a given Ethereum address has a Farcaster user associated with it.
+   * Note: if an address is associated with multiple users, the API will return
+   * the user who most recently published a verification with the address
+   * (based on when Merkle received the proof, not a self-reported timestamp).
+   */
+  public async lookupUserByVerification(
+    address: string
+  ): Promise<User | undefined> {
+    try {
+      const response = await this.apis.verification.userByVerification(address);
+      return response.data.result?.user?.fid != null
+        ? response.data.result.user
+        : undefined;
+    } catch (error) {
+      if (NeynarAPIClient.isApiErrorResponse(error)) {
+        if (error.response.status === 404) return undefined;
+      }
+      throw error;
+    }
+  }
+
+  public async *fetchMentionAndReplyNotifications(
+    fid: number,
+    { pageSize = 100 } = {}
+  ): AsyncGenerator<v1Cast, void, undefined> {
+    let cursor: string | undefined;
+    let viewer: number | undefined;
+
+    while (true) {
+      // fetch one page of notifications
+      const response = await this.apis.notifications.mentionsAndReplies(
+        fid,
+        viewer,
+        cursor,
+        pageSize
+      );
+
+      // yield current page
+      yield* response.data.result.notifications;
+
+      // prep for next page
+      if (response.data.result.next?.cursor === undefined) {
+        break;
+      }
+      cursor = response.data.result.next.cursor ?? undefined;
+    }
+  }
+
+  /**
+   * Lists a given cast's likes
+   */
+  public async *fetchCastLikes(
+    castOrCastHash: v1Cast | string,
+    { pageSize = 100 } = {}
+  ): AsyncGenerator<Reaction, void, undefined> {
+    let cursor: string | undefined;
+    let viewer: number | undefined;
+    let castHash: string;
+    if (typeof castOrCastHash === "string") {
+      castHash = castOrCastHash;
+    } else {
+      castHash = castOrCastHash.hash;
+    }
+
+    while (true) {
+      const response = await this.apis.reactions.castLikes(
+        castHash,
+        viewer,
+        cursor,
+        pageSize
+      );
+
+      yield* response.data.result.likes;
+
+      // prep for next page
+      if (response.data.result.next?.cursor === undefined) {
+        break;
+      }
+      cursor = response.data.result.next.cursor ?? undefined;
+    }
+  }
+
+  /**
+   * Get all users that follow the specified user
+   */
+  public async fetchUserFollowers(user: {
+    fid: number;
+  }): Promise<User[] | undefined> {
```

Comment:
What's the significance of an empty list here vs an undefined return value? Same question for `fetchUserFollowing`

Reply:
I removed undefined, it only returns []